sonic_y_cable/y_cable_base.py

```diff
@@ -0,0 +1,577 @@
+
+            lane_map:
+                 an Integer, representing the lane_map to be run PRBS on
+                 0bit for lane 0, 1bit for lane1 and so on.
+                 for example 3 -> 0b'0011 , means running on lane0 and lane1
+
+        Returns:
+            a boolean, true if the enable is successful
+                     , false if the enable failed
+
+        """
+
+        raise NotImplementedError
+
+    def disable_prbs_mode(self, port, target):
+        """
+        This API specifically disables the PRBS mode on the physical port.
+
+        Args:
+            physical_port:
+                 an Integer, the actual physical port connected to a Y cable
+            target:
+                 an Integer, the target on which to enable the PRBS
+                             EYE_PRBS_TARGET_LOCAL -> local side,
+                             EYE_PRBS_TARGET_TOR1 -> TOR 1
+                             EYE_PRBS_TARGET_TOR2 -> TOR 2
+                             EYE_PRBS_TARGET_NIC -> NIC
+
+        Returns:
+            a boolean, true if the disable is successful
+                     , false if the disable failed
+        """
+
+        raise NotImplementedError
+
+    def enable_loopback_mode(self, port, target, lane_map):
+        """
+        This API specifically configures and enables the Loopback mode on the port user provides.
+        Target is an integer for selecting which end of the Y cable we want to run loopback on.
+        LaneMap specifies the lane configuration to run the loopback on.
+        Note that this is a diagnostic mode command and must not run during normal traffic/switch operation
+
+        Args:
+            physical_port:
+                 an Integer, the actual physical port connected to a Y cable
+            target:
+                 an Integer, the target on which to enable the PRBS
+                             EYE_PRBS_TARGET_LOCAL -> local side,
+                             EYE_PRBS_TARGET_TOR1 -> TOR 1
+                             EYE_PRBS_TARGET_TOR2 -> TOR 2
+                             EYE_PRBS_TARGET_NIC -> NIC
+            lane_map:
+                 an Integer, representing the lane_map to be run PRBS on
+                 0bit for lane 0, 1bit for lane1 and so on.
+                 for example 3 -> 0b'0011 , means running on lane0 and lane1
+
+        Returns:
+            a boolean, true if the enable is successful
+                     , false if the enable failed
+        """
+
+        raise NotImplementedError
+
+    def disable_loopback_mode(self, port, target):
+        """
+        This API specifically disables the Loopback mode on the port user provides.
+        Target is an integer for selecting which end of the Y cable we want to run loopback on.
+
+        Args:
+            physical_port:
+                 an Integer, the actual physical port connected to a Y cable
+            target:
+                 an Integer, the target on which to enable the PRBS
```

Comment:
Maybe rather than explicitly stating "An Integer", we simply state "One of the following predefined constants:". This would also allow us to change the underlying type transparently if desired.

Reply:
fixed